### PR TITLE
Add 'matb33:collection-hooks' type definitions

### PR DESCRIPTION
**Pull request by Urigo**
# Meteor Collection Hooks Definitions

Type definitions for Meteor package `matb33:collection-hooks`.
Be sure to reference any sort of basic Meteor type definitions file before use.

### CODE REVIEW

File: meteor-collection-hooks/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
+// Type definitions for Meteor package matb33:collection-hooks
+// Project: https://github.com/matb33/meteor-collection-hooks
+// Definitions by: Trygve Wastvedt <https://github.com/twastvedt>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module 'meteor/mongo' {
```
Comment (DanielRosenwasser) [inline meteor-collection-hooks/index.d.ts]: Is this supposed to augment `meteor/mongo`?

Comment (Urigo) [inline meteor-collection-hooks/index.d.ts]: Indeed. There should be two declarations (From Meteor@1.3.x onwards): One for the `Mongo` object defined on the global scope, and the second for the `Mongo` object defined in the `meteor/mongo` module.

**Comment by DAB0mB** [comment]
@DanielRosenwasser is there anything new with this PR? I actually really need it...

**Comment by aozgaa** [comment]
Could you add a `tslint.json`? Sorry for the delays.

File: meteor-collection-hooks/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  interface Selector {}
+  interface Modifier {}
+  interface ObjectID {}
+  interface SortSpecifier {}
+  interface FieldSpecifier {}
+  interface Cursor<T> {}
+
+  interface Collection<T> {
+    before: {
+      find(hook: {(userId: string, selector: Mongo.Selector, options: { multi?: boolean; upsert?: boolean; }): void}): void;
```
Comment (aozgaa) [inline meteor-collection-hooks/index.d.ts]: Can you consolidate the `before`/`after` types into one named interface?

Comment (aozgaa) [inline meteor-collection-hooks/index.d.ts]: So as to avoid duplicating code?

File: meteor-collection-hooks/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      find(hook: {(userId: string, selector: Mongo.Selector, options: { multi?: boolean; upsert?: boolean; }, cursor: Mongo.Cursor<T>): void}): void;
+      findOne(hook: {(userId: string, selector: Mongo.Selector, options: { multi?: boolean; upsert?: boolean; }, doc: T): void}): void;
+      insert(hook: {(userId: string, doc: T): void}): void;
+      remove(hook: {(userId: string, doc: T): void}): void;
+      update(hook: {(userId: string, doc: T, fieldNames: string[], modifier: Mongo.Modifier, options: { multi?: boolean; upsert?: boolean; }): void}, options?: HookOptions): void;
+      upsert(hook: {(userId: string, doc: T, selector: Mongo.Selector, modifier: Mongo.Modifier, options: { multi?: boolean; upsert?: boolean; }): void}): void;
+    };
+    direct: {
+      find(selector?: Mongo.Selector | Mongo.ObjectID | string, options?: {
+        sort?: Mongo.SortSpecifier;
```
Comment (aozgaa) [inline meteor-collection-hooks/index.d.ts]: Can you make the `options` object into a named interface, as above?

File: meteor-collection-hooks/index.d.ts (hunk 2 above)
Comment (aozgaa) [inline meteor-collection-hooks/index.d.ts]: Can you also factor out the `options` type into some named interface?

File: meteor-collection-hooks/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  }
+}
+
+declare interface HookOptions {
+  fetchPrevious?: boolean;
+}
+
+declare interface CollectionOptions {
+  before?: {
+    all?: HookOptions;
```
Comment (aozgaa) [inline meteor-collection-hooks/index.d.ts]: Can you make this object's type some named interface?

**Comment by Urigo** [comment]
@aozgaa @DanielRosenwasser I've just finished going through things you've pointed out in your review. Hopefully it's OK now

**Pull request merged**